front/src/applications/stdcm/components/StdcmForm/StdcmConsist.tsx

```diff
@@ -266,19 +269,26 @@ const StdcmConsist = ({
         />
       </div>
       <div className="loading-gauge">
-        <Select
-          id="loading-gauge-selector"
-          value={loadingGauge}
-          label={t('consist.loadingGauge')}
-          onChange={(e) => {
-            if (e) {
-              onLoadingGaugeChange(e);
-            }
-          }}
-          {...createStandardSelectOptions(GAUGE_LIST)}
-          disabled={disabled}
-          narrow
-        />
+        {loadingGauges && loadingGauges.length > 0 && (
+          <Select
+            id="loading-gauge-selector"
+            value={loadingGauge}
+            label={t('consist.loadingGauge')}
+            onChange={(e) => {
+              if (
+                e &&
+                ['G1', 'G2', 'GA', 'GB', 'GB1', 'GC', 'FR3.3', 'FR3.3/GB/G2', 'GLOTT'].includes(e)
+              ) {
+                onLoadingGaugeChange(e as LoadingGaugeType);
+              } else {
+                onLoadingGaugeChange(undefined);
+              }
+            }}
```

Comment:
To keep some type verification, I would still keep the pre-existing `const` on top of the component in case the api type changes in the future : 
```
const GAUGES_LIST: LoadingGaugeType[] = ['G1', 'G2', 'GA', 'GB', 'GB1', 'GC', 'FR3.3', 'FR3.3/GB/G2', 'GLOTT'];
const GAUGES_SET = new Set<string>(GAUGES_LIST);
const isLoadingGaugeType = (gauge: string | undefined): gauge is LoadingGaugeType =>
  typeof gauge === 'string' && GAUGES_SET.has(gauge);
```

and here you could do : 


```suggestion
            onChange={(e) => {
              if (isLoadingGaugeType(e)) {
                onLoadingGaugeChange(e);
              } else {
                onLoadingGaugeChange(undefined);
              }
            }}
```

Benefits : 
- type control on the received strings
- no cast (we try to avoid them in the front code base) in the `onChange` because of the type guard `isLoadingGaugeType`

Reply:
Done